### DISCUSSION TITLE
What's the best way to declare an empty SafeHandle?

### DISCUSSION THREAD

**Discussion by HavenDV** [microsoft/CsWin32#516] [Q&A] [answered]
What's the best way to declare an empty SafeHandle when this parameter is ignored (for example, in CallNextHookEx)?

**Comment by jnm2** [comment, marked as answer]
I believe you can pass `null` to the friendly overload which will handle it correctly. (Unlike how traditional p/invoke has worked.)

**Reply by HavenDV** [reply]
Thanks, in these methods it is really possible to pass `null` to the SafeHandle place and it will be correctly processed (an empty structure will be passed). I guess I didn't notice it because of this issue (I'm already so used to nullable annotations)
https://github.com/microsoft/CsWin32/issues/130

**Reply by AArnott** [reply]
In addition, if calling the extern method itself, you can pass in `default` as the value so the native code gets a 0.
